sdk/identity/azure-identity/test/perf/inc/azure/identity/test/secret_credential_test.hpp

```diff
@@ -41,7 +41,10 @@ namespace Azure { namespace Identity { namespace Test {
       m_clientId = m_options.GetMandatoryOption<std::string>("ClientId");
       m_secret = m_options.GetMandatoryOption<std::string>("Secret");
       m_tokenRequestContext.Scopes.push_back(m_options.GetMandatoryOption<std::string>("Scope"));
-      m_tokenRequestContext.CacheCredentials = m_options.GetOptionOrDefault<bool>("Cache", false);
+      if (m_options.GetOptionOrDefault<bool>("Cache", false) == false)
```

Comment:
Out of curiosity, why the explicit check against `false`?

Why not:

```cpp
if (!m_options.GetOptionsOrDefault<bool>("Cache", false))
{
```

I'm not opposed to the code, but it seems redundant and possibly fragile.

Reply:
isn't it the same thing ? i know i know one is a logical ! and compare with true the other one is a compare with 0

Reply:
It is the same thing. It's just unusual to test a boolean value for equality against another boolean rather than simply testing the boolean value.

In other words, it's the difference between:

```cpp
if (val == true)
```

and

```cpp
if (val)
```

The first looks slightly weird to me. But it's a personal style thing.